modules/bidViewabilityIO.js

```diff
@@ -0,0 +1,87 @@
+import { config } from '../src/config.js';
+import * as events from '../src/events.js';
+import { EVENTS } from '../src/constants.json';
+import { logMessage } from '../src/utils.js';
+
+const MODULE_NAME = 'bidViewabilityIO';
+const CONFIG_ENABLED = 'enabled';
+
+// IAB numbers from: https://support.google.com/admanager/answer/4524488?hl=en
+const IAB_VIEWABLE_DISPLAY_TIME = 1000;
+const IAB_VIEWABLE_DISPLAY_LARGE_PX = 242000;
+export const IAB_VIEWABLE_DISPLAY_THRESHOLD = 0.5
+export const IAB_VIEWABLE_DISPLAY_LARGE_THRESHOLD = 0.3;
+
+const CLIENT_SUPPORTS_IO = window.IntersectionObserver && window.IntersectionObserverEntry && window.IntersectionObserverEntry.prototype &&
+    'intersectionRatio' in window.IntersectionObserverEntry.prototype;
+
+const supportedMediaTypes = [
+  'banner'
+];
+
+export let isSupportedMediaType = (bid) => {
+  return supportedMediaTypes.indexOf(bid.mediaType) > -1;
+}
+
+// returns options for the iO that detects if the ad is viewable
+export let getViewableOptions = (bid) => {
+  if (bid.mediaType === 'banner') {
+    return {
+      root: null,
+      rootMargin: '0px',
+      threshold: bid.width * bid.height > IAB_VIEWABLE_DISPLAY_LARGE_PX ? IAB_VIEWABLE_DISPLAY_LARGE_THRESHOLD : IAB_VIEWABLE_DISPLAY_THRESHOLD
+    }
+  }
+}
+
+// markViewed returns a function what will be executed when an ad satisifes the viewable iO
+export let markViewed = (bid, entry, observer) => {
+  return () => {
+    observer.unobserve(entry.target);
+    events.emit(EVENTS.BID_VIEWABLE, bid);
+    logMessage(`id: ${entry.target.getAttribute('id')} code: ${bid.adUnitCode} was viewed`);
```

Comment:
This is me being picky, but when I am testing / debugging pub pages I like when modules emit their name inside their console logs, makes it easy to filter.

You could wrap logMessage in your own function which just adds in MODULE_NAME on each message you send.

Reply:
That doesn't sound picky, it sounds like an excellent practice.